cmd/fmt.go

```diff
@@ -126,13 +127,21 @@ func formatFile(params *fmtCommandParams, out io.Writer, filename string, info o
 		return newError("failed to open file: %v", err)
 	}
 
-	opts := format.Opts{}
-	opts.RegoVersion = params.regoVersion()
+	opts := format.Opts{
+		RegoVersion: params.regoVersion(),
+	}
 	formatted, err := format.SourceWithOpts(filename, contents, opts)
 	if err != nil {
 		return newError("failed to format Rego source file: %v", err)
 	}
 
+	if params.checkResult {
+		_, err := ast.ParseModule("formatted", string(formatted))
```

Comment:
Would we need to compile the module too, in order to ensure it's valid?

Reply:
We could do that 🤔. `SourceWithOpts` will however run all v1-specific compiler checks.

We'd probably need to turn off a bunch of stuff in the compiler, though. Just because someone is formatting their Rego doesn't necessarily mean the code is completely valid .. Things that the parser allows might be rejected by the compiler.